syft/federated/plan.py

```diff
@@ -314,12 +351,29 @@ def __call__(self, *args, **kwargs):
             The pointer to the result of the execution if the plan was already sent,
             else the None message serialized.
         """
-        assert len(kwargs) == 0, "kwargs not supported for plan"
+        if len(kwargs):
+            raise ValueError("Kwargs are not supported for plan.")
+
+        # TODO: for now only one value is returned from a plan
         result_ids = [sy.ID_PROVIDER.pop()]
+
+        # TODO: try to find a better way to deal with local execution
+        # of methods.
+        if self.is_method and not self.locations and self.owner == sy.hook.local_worker:
```

Comment:
What conditions result in the changing of the values of these checks

Reply:
This hack is needed to make a method (aka a module) be executed locally in a transparent way. Mainly this will not be true if the plan was sent to some location.

Reply:
So for clarification. Self.locations is set when send is called. Self.owner is set when?

Reply:
Is the self.owner part of this check necessary then?

Reply:
It’s a useful hack but it only works for virtual workers 
We need a better way to know if the worker is the local worker (aka client)

Reply:
Would it work if it was sent? does it keep the same reference to the local worker after it is sent over the wire?

Reply:
> Would it work if it was sent? does it keep the same reference to the local worker after it is sent over the wire?

Considering that it was already built. Yes, it will work.

No, it doesn't keep the same reference.

But as @LaRiffle said we consider that the blueprint owner is a virtual worker which is usually true since when a model is instantiated it belongs to hook.local_worker which is a virtual worker. So I don't think we need to worry about this condition right now.